controllers/azurecluster_reconciler_test.go

```diff
@@ -291,15 +300,14 @@ func TestAzureClusterServiceDelete(t *testing.T) {
 			expectedError: "failed to delete AzureCluster service two: some error happened",
 			clientBuilder: func(g Gomega) client.Client {
 				scheme := runtime.NewScheme()
+				g.Expect(infrav1.AddToScheme(scheme)).To(Succeed())
 				g.Expect(asoresourcesv1.AddToScheme(scheme)).To(Succeed())
 
 				rg := &asoresourcesv1.ResourceGroup{
 					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceGroup,
-						Namespace: namespace,
-						Labels: map[string]string{
-							infrav1.OwnedByClusterLabelKey: "not-" + clusterName,
-						},
```

Comment:
Should we keep a test case that uses the legacy label to test back compat?

Reply:
That scenario is already tested in the `aso` package, so I'm not sure if it's worth duplicating that here.